### DISCUSSION TITLE
Port Checking

### DISCUSSION THREAD

**Discussion by Soneji** [upptime/upptime#204] [Ideas] [answered]
Hi

I was wondering if there's a way to check for a specific port, not a HTTP service. For example, uptimerobot.com can check if a port closes, which would mean a service, such as a minecraft instance, was to go down. Is there any way to do this with upptime? 

<img width=400 src=https://user-images.githubusercontent.com/12625848/104136546-dc373880-538e-11eb-9c20-19cb673f3238.png>

**Comment by AnandChowdhary** [comment]
I think you can just do `https://example.com:23891`, adding the port to the URL.

**Comment by Soneji** [comment]
That would certainly work for an HTTP service on a separate port but I was thinking something more broad, which could check for open ports for any service type. Unfortunately I don’t think that would work for the Minecraft server since it does not use HTTP

**Reply by AnandChowdhary** [reply]
It seems like this is possible, I'll look into it! https://stackoverflow.com/q/20434620/1656944

**Reply by Soneji** [reply]
Maybe this package could be handy

https://www.npmjs.com/package/is-port-reachable

**Reply by thomasmerz** [reply]
@AnandChowdhary , with this speciific curl from StackOverflow it results in `curl: (1) Received HTTP/0.9 when not allowed`, when curling a MineCraft-Server IP:Port… 🤔 
@soneji , this `npm is-port-reachable` looks great - but unfortunately I'm no programmer 😢

**Reply by thomasmerz** [reply]
Do you @AnandChowdhary have any news regarding port-checking for us? 🤔 See also #238

**Reply by Soneji** [reply]
@AnandChowdhary any update on this?

**Comment by AnandChowdhary** [comment, marked as answer]
Support for pinging using [`tcp-ping`](https://github.com/apaszke/tcp-ping) has been added in [@upptime/uptime-monitor@v1.25.1](https://github.com/upptime/uptime-monitor/releases/tag/v1.25.1).

You'll need to add the property `check: "tcp-ping"` to use this:

```yaml
sites: # List of endpoints to track
  - name: Google
    check: "tcp-ping"
    url: 46.28.246.123
    port: 80
```

**Reply by Soneji** [reply]
nice, thank you! :)

**Reply by Soneji** [reply]
do you mean 1.25.0?
![image](https://user-images.githubusercontent.com/12625848/109004599-de382b00-76a0-11eb-9adb-af1750225c40.png)

**Reply by AnandChowdhary** [reply]
No, the version is currently being released, I just commented a few minutes before it's up! 😄

**Reply by Soneji** [reply]
Ok, thanks! Do you know how long it will take to be avalable to the updates CI?

**Reply by AnandChowdhary** [reply]
It's ready now!

**Reply by Soneji** [reply]
![image](https://user-images.githubusercontent.com/12625848/109033187-49dcc100-76be-11eb-9aa3-afeb9ffd8cc9.png)
I am getting this error, is that because it is the first time running it?

**Reply by AnandChowdhary** [reply]
Yes, that could be the case, or because the `avg` value is NaN for some reason. 😕

**Reply by AnandChowdhary** [reply]
Ok, I see that the `tcp-ping` is throwing some sort of error, I'll have a detailed look!

**Reply by Soneji** [reply]
Thank you :)

**Reply by Soneji** [reply]
Heya

Just wondering if you managed to fix this?

**Reply by timbze** [reply]
> Support for pinging using [`tcp-ping`](https://github.com/apaszke/tcp-ping) has been added in [@upptime/uptime-monitor@v1.25.1](https://github.com/upptime/uptime-monitor/releases/tag/v1.25.1).
> 
> You'll need to add the property `check: "tcp-ping"` to use this:
> 
> ```yaml
> sites: # List of endpoints to track
>   - name: Google
>     check: "tcp-ping"
>     url: 46.28.246.123
>     port: 80
> ```

Seems like this needs a documentation update. I was not able to set up ping till I found this. When I add `check: "tcp-ping"`, it works
